docs/DeveloperGuide.md

```diff
@@ -158,10 +158,46 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+`GroupedUniquePersonList` adds a new string called `groupName` to label each of their groups.
+
+A new list of `GroupedUniquePersonList` will be added to the `Model` interface.
+
+To add to the list of `GroupedUniquePersonList`, the Model interface includes `Model#addGroup(String, List<Person>)` and `Model#addListOfGroups(List<Group>)`.
+
+To remove to the list of `GroupedUniquePersonList`, the Model interface includes `Model#removeGroup(String)`.
+
+New operation are exposed in the `Model` interface are `Model#addPersonToGroup(String, Person)`, `Model#removePersonFromGroup(String, Person)` and `Model#deleteAssignedTaskGroup(String, Task)` which would call `UniquePersonList#add(Person)`, `UniquePersonList#remove(Person)` and `UniquePersonList#deleteAssignedTask(Person)`respectively.
+
+Given below is an example usage scenario and how the grouping mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The list of the `GroupedUniquePersonList` will be empty if there are no groups stored in the storage.
+
+Step 2. The user executes `group gn/2103T gp/1 gp/2 gp/3` command to group person 1, 2 and 3 from the displayed person list to one group. The `group` command calls `Model#addGroup(String, List<Person>)`, which creates a new group with that contains the list of people that was indicated by the user.
+
+Step 3. The user executes `assigngroup gn/2103T t/1` command to assign a task to the group named "2103T" from the group list. The `assigngroup` command calls `Model#assignTaskToGroup(String, Task)`, which finds the group with the same name and assign that task to everyone that is in the group.
+
+Step 4. The user executes `addpersontogroup gn/2103T gp/4` command to add a new person 4 to the group named "2103T" from the group list. The `addpersontogroup` command calls `Model#addPersonToGroup(String, Person)`, which finds the group with the same name and add the person to the group.
+
+Step 5. The user executes `removepersonfromgroup gn/2103T gp/4` command to remove person 4 to the group named "2103T" from the group list. The `removepersonfromgroup` command calls `Model#removePersonFromGroup(String, Person)`, which finds the group with the same name and remove the person to the group.
+
+Step 6. The user executes `deletetaskgroup gn/2103T` command to remove a task from the group named "2103T" from the group list. The `deletetaskgroup` command calls `Model#deleteAssignedTaskGroup(String, Task)`, which finds the group with the same name and remove that task from everyone that is in the group.
+
+Step 7. The user executes `deletegroup gn/2103T` command to remove the group from the list. The `deletegroup` command calls `Model#removeGroup(String)`, which finds the group with the same name and remove that group from the list.
+
```

Comment:
Maybe you can describe how the UI will show the group(s) a person is currently in, and briefly describe how the UI will change when each command related to grouping is executed.